spark/src/main/scala/org/apache/spark/sql/delta/DeltaOperations.scala

```diff
@@ -75,9 +76,12 @@ object DeltaOperations {
     }
   }
 
-  abstract class OperationWithPredicates(name: String, val predicates: Seq[Expression])
+  abstract class OperationWithPredicates(name: String, val predicates: Seq[Expression]
+                                         , metadata: Metadata)
       extends Operation(name) {
-    private val predicateString = JsonUtils.toJson(predicatesToString(predicates))
+    private val predicateString = JsonUtils.toJson(predicatesToString(predicates).map(
+        c => StringUtils.abbreviate(c,
+          DeltaConfigs.MAX_AMOUNT_OF_CHAR_IN_TRANSACTION_PREDICATE.fromMetaData(metadata))))
```

Comment:
what is the reason to make this configurable in the table? If we imagine that write operations fail when serializing a predicate, then it would be better if this is a dynamic session config (like spark sql conf)

Reply:
But what if I want different numbers of elements for different tables in the same session. for that metadata is better

Reply:
Sessions are also not permanent (would not apply to all workloads). I think you are right, metadata makes more sense for now